Feature/#1: 회원가입/로그인 컴포넌트 및 스토리북 구현

minjeongss opened the pull request:
## #️⃣ 연관된 이슈

#1 

## 📝 작업 내용
- 회원가입/로그인 화면에서 사용되는 공통 입력창 컴포넌트 구현 및 스토리북 구현
- 회원가입 화면에서 사용되는 닉네임 입력 컴포넌트 구현 및 스토리북 구현


## 💬 리뷰 요구사항
- 공통 컴포넌트가 아닌 컴포넌트의 파일 구조 방식이 올바른지 확인 부탁드립니다.
  - 위치는 다음과 같습니다. 
  ``` 
  app/signup/components/*
  ```
- 스토리북 사용 방법 중 잘못된 부분이 있는지 확인 부탁드립니다.
  - args에서 title, placeholder와 같은 부분을 변경할 수 있도록 해야 할까요?
  -  args에서 handleClick의 내부 내용은 어떻게 채워야 할까요?

minjeongss committed:
feat: 입력창 컴포넌트 구현(닉네임 제외)

minjeongss committed:
feat: 닉네임 입력창 컴포넌트 구현

- 입력창은 Input 컴포넌트 활용
- 중복확인 버튼은 signup 내부의 components에서 작성

minjeongss committed:
feat: 입력창 스토리북 구현

- onClick event 제외

minjeongss committed:
feat: 닉네임 입력창 스토리북 구현

- onClick 이벤트 제외 구현

minjeongss committed:
feat: storybook example, Input 폴더 분리

- example: 기본으로 제공해주는 파일들 묶음
- Input: 회원가입+로그인에서 활용되는 입력창 묶음

minjeongss committed:
feat: Input 컴포넌트에 handleClick 이벤트 추가

minjeongss committed:
feat: NickNameInput 컴포넌트 handleClick 구현

- 단순 alert창 추가
- 동작 연결은 기능 구현에서 수행

minjeongss committed:
fix: styled component에서 prop이 실제 dom에 전달되어 warning 발생하는 문제 해결

- 기존 방식: styled component의 prop을 isHiddenPassword로 사용
- 변경된 방식: styled component의 prop을 로 사용해, prop을 스타일링 목적으로만 사용하도록 설정

minjeongss committed:
style: Signup 화면에서 안내 문구 margin 부여

aaahyesu commented on stories/Input/Input.stories.ts:
```diff
@@ -1,5 +1,5 @@
+import Input from "../../components/Input/Input"; // 수정된 부분
 
 const meta = {
   title: "Input/Input",
```
이 부분이 스토리북에서의 폴더 구조인데, 저 같은 부분에는 `title: "Common/Button/CommonButton"`으로 진행했었습니다..!
이 폴더 구조를 맞춰보면 어떨까하는데... 


**1. 뎁스를 한단계 공통/기능별로(`Common/Input` , `Layout/Login`) 들어간다.
2. 하나에 다 넣는게 좋다. (`Input`, `Login`)**

둘 중에 어떤게 더 나을지 의견을 알려주시면 좋을 것 같아요-! 👾

minjeongss replied:
어렵네요.. 제가 Input에 넣는 부분은
- Input/Input.stories.ts
- Input/NickNameInput.stories.ts
인데!

여기에서 Input.tsx는 root/components 내부에 존재하지만, NickNameInput.tsx는 app/signup/components 내부에 존재하거든요..!
저 Common과 Common이 아닌 것에 대한 분리를 의논해봐야 할 것 같아요~

minjeongss replied:
특정 기능에 대한 스토리가 아니면 다 Common에 넣고, 기능에 대한 부분은 Layout에 넣는 방식으로 가기로 결정 ! 👍

aaahyesu commented on stories/Input.stories.ts:
```diff
@@ -0,0 +1,27 @@
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const NormalContent: Story = {
```
> args에서 title, placeholder와 같은 부분을 변경할 수 있도록 해야 할까요?

이 부분을 storybook에서 모든것이 적용된 컴포넌트가 아닌 직접 수정을 해보면서 확인할 수 있게 끔 만들어야 되는가..? 라고 이해했습니다!

이 점이 맞다면 제 생각으로는 "이 컴포넌트가 다른 곳에서도 쓰이는가?"를 생각해보고 그게 맞다라고 한다면 변경할 수 있도록 하고, 그렇지 않다면 간단하게 확인만 하고 넘어갈 수 있도록 하는게 맞다고 생각합니다아-! 👏

minjeongss replied:
아하 컴포넌트 자체를 사용할 때 prop으로 title, placeholder와 같은 부분은 입력을 해야 하는데, 이를 storybook에선 따로 입력받지 않도록 구현을 했었습니다.

그러면 storybook에도 반영이 될 수 있도록 해볼게요!

aaahyesu commented on stories/Input/Input.stories.ts:
```diff
@@ -18,10 +18,22 @@ export const NormalContent: Story = {
+    placeholder: "8자 이상의 문자 입력 (숫자, 영어, 특수문자 포함)",
+    isButton: true,
+    isHiddenPassword: false,
+    handleClick: () => {},
```
> args에서 handleClick의 내부 내용은 어떻게 채워야 할까요?

저는 이렇게만 해도 충분하다고 생각합니다 고생하셨어요!🤍

minjeongss committed:
feat: Signup 페이지에서 비밀번호 보여주는 기능 독립적 작동 구현

- 비밀번호와 비밀번호 확인에서 비밀번호 보여주는 기능 각각에서 따로 작동하도록 구현

bluedog129 reviewed (approved):
로그인 컴포넌트 및 스토리북 구현 수고 하셨습니다!!

minjeongss committed:
feat: story book 폴더 구조 변경

- 기능이 아닌 컴포넌트: Common 내부에 위치
- 기능인 컴포넌트: Layout 내부에 위치

minjeongss committed:
style: 회원가입 입력창 사이 공백 스타일 정렬

- 입력창 margin 사용하자 겹치는 문제 존재해서 padding으로 변경
- signup의 title은 margin을 그대로 사용

minjeongss committed:
feat: Signup 로직 폴더 구조 변경
>
> - app/signup/components > components/Signup으로 변경
- stories/Input/NickNameInput > stories/Signup/NickNameInput으로 변경
